tests/ui_tools/test_boxes.py

```diff
@@ -822,19 +824,24 @@ def test_keypress_ENTER(self, panel_search_box, widget_size,
         # Update this display
         # FIXME We can't test for the styled version?
         # We'd compare to [('filter_results', 'Search Results'), ' ']
-        assert panel_search_box.caption == self.search_caption
-        assert panel_search_box.edit_text == "key words"
 
-        # Leave editor mode
-        (panel_search_box.panel_view.view.controller.exit_editor_mode
-         .assert_called_once_with())
+        assert panel_search_box.edit_text == "key words"
 
-        # Switch focus to body; if have results, move to them
-        panel_search_box.panel_view.set_focus.assert_called_once_with("body")
         if expect_body_focus_set:
+            assert panel_search_box.caption == self.search_caption
+            # Leave editor mode
+            (panel_search_box.panel_view.view.controller.exit_editor_mode
+             .assert_called_once_with())
+            # Switch focus to body; if have results, move to them
+            panel_search_box.panel_view.set_focus.assert_called_once_with(
+                "body")
             (panel_search_box.panel_view.body.set_focus
              .assert_called_once_with(0))
         else:
+            assert panel_search_box.caption == ''
+            (panel_search_box.panel_view.view.controller.exit_editor_mode
+             .assert_not_called())
+            panel_search_box.panel_view.set_focus.assert_not_called()
```

Comment:
With these changes and large conditional branches, I'd suggest splitting the parametrize and having two separate test functions. If you do so, please have it in a separate commit first :+1:

Reply:
Hmm, ok I'll split the test :+1:

Reply:
You should be able to remove the parametrize completely and do some other simplification in each test now.